lib/src/main/java/io/ably/lib/types/BaseMessage.java

```diff
@@ -93,9 +98,13 @@ public void decode(ChannelOptions opts) throws AblyException {
 						catch(JsonParseException e) { throw AblyException.fromThrowable(e); }
 						continue;
 					}
-					if(xform == "cipher" && opts != null && opts.encrypted) {
-						data = opts.getCipher().decrypt((byte[])data);
-						continue;
+					if(xform == "cipher") {
+						if (opts != null && opts.encrypted) {
+							data = opts.getCipher().decrypt((byte[]) data);
+							continue;
+						}
+						else
```

Comment:
If we have a compound if/then/else then it either needs braces everywhere, or nowhere, but not mixed like this. In in doublt, use braces.